RNUnifiedContacts/RNUnifiedContacts.swift

```diff
@@ -163,17 +163,94 @@ class RNUnifiedContacts: NSObject {
 
       callback([error.localizedDescription, NSNull()])
     }
-
   }
-
-
-
-
-
+  
+  @objc func createContact(contactData: NSDictionary, callback: (NSObject) -> () ) -> Void {
+    
+    let mutableContact = CNMutableContact()
+    let contactStore = CNContactStore()
+    let req = CNSaveRequest()
+    
+    // TODO:
+    // Extend method to handle more fields
+    //
+    //
+    mutableContact.givenName = contactData["givenName"] as! String
+    mutableContact.familyName = contactData["familyName"] as! String
+    mutableContact.organizationName = contactData["organizationName"] as! String
+    
+    for number in contactData["phoneNumbers"] as! NSArray {
+      let transformedNumber = transformNumber( number as! NSDictionary )
+      mutableContact.phoneNumbers.append( transformedNumber )
+    }
+    
+    for email in contactData["emailAddresses"] as! NSArray {
+      let transformedEmail = transformEmail ( email as! NSDictionary )
+      mutableContact.emailAddresses.append( transformedEmail )
+    }
+    
+    do {
+      req.addContact(mutableContact, toContainerWithIdentifier:nil)
+      try contactStore.executeSaveRequest(req)
+      print("Successfully created contact")
+      callback( [NSNull(), ""] )
+    } catch let error as NSError {
+      print("Something went wrong")
+      callback( [error.localizedDescription, NSNull()] )
+    }
+    
+  }
+  
+  @objc func deleteContact(identifier: String, callback: (NSObject) -> () ) -> Void {
+    
+    let contactStore = CNContactStore()
+    
+    let cNContact = getCNContact( identifier, keys: keysToFetch )
+    let req = CNSaveRequest()
+    let mutableContact = cNContact!.mutableCopy() as! CNMutableContact
+    req.deleteContact(mutableContact)
+    
+    do {
+      
+      try contactStore.executeSaveRequest(req)
+      NSLog("Success, You deleted the user with identifier: " + identifier)
+      
+      callback( [NSNull(), identifier] )
```

Comment:
Well basically it seems to be an event triggered everytime something changes in the contact book. Would be handy to have in RN. Primary usage would be to have synced records.